### ISSUE TITLE
Are pension benefit derivatives correct in Euler equations (code and docs)?

### ISSUE THREAD

**Issue by rickecon** [PSLmodels/OG-Core#980]
In going through the documentation for the new pension module and options in OG-Core, it strikes me that the derivatives of the replacement rates with respect to labor supply or savings are not in any standard forms of our household first order conditions. However, they do show up in the `stationarization.md` chapter, but only in the "Stationarized pension system" equations, but not in the "Stationarized household equations" nor in the "Government" chapter.

I want to make sure that these derivatives are correctly accounted for in the theory, documentation, and code.

cc: @jdebacker

**Comment by jdebacker** [comment]
@rickecon You are correct, the derivatives of the replacement rates are NOT in the household FOCs.  

They are specified in `pensions.py`.  They are also defined in the "Stationarization" and "Government" chapters of the documentation (though they are left out of the household equations in "Stationarization" and "Households" as they are left out of the code in `households.py`).  The exception to this is that the derivative of the replacement rate is not specified in the docs or `pensions.py` for the US-style SS system. 

I think the theory is correct (at least we had several minds thinking about this in 2019-2020) and the equations in `pensions.py` are tested.  But I would love to get a review of this from @jpycroft.

In PR #946, I omitted using these derivatives in the household FOCs because:
1. We have yet to specify a similar equations for the US Style SS system (and given how OG-Core currently compute those benefits, it's not straight forward).
2. A cost-benefit analysis: I see the stochastic income on the horizon, I don't see too many use cases for the other pension systems before then -- and that new household solution method with stochastic income will likely involve some changes to how we solve for pensions.  Also, I think the derivatives or replacement rates don't play a quantitatively significant role in the household decision for most agents in most periods given the time discounting between the labor supply decision and the receipt of pension benefits.  Thus the benefits of implementing these over the time horizon between now and a major refactoring didn't outweigh the costs of doing so.
